Nodejs/Product/Nodejs/NpmUI/NpmPackageInstallViewModel.cs

```diff
@@ -357,17 +367,64 @@ public PackageCatalogEntryViewModel SelectedPackage
             }
         }
 
+        public bool IsExecutingCommand
+        {
+            get
+            {
+                return this.isExecutingCommand;
+            }
+            set
+            {
+                this.isExecutingCommand = value;
+
+                OnPropertyChanged();
+                OnPropertyChanged("ExecutionProgressVisibility");
+            }
+        }
+
+        public Visibility ExecutionProgressVisibility
+        {
+            get { return IsExecutingCommand ? Visibility.Visible : Visibility.Collapsed; }
+        }
+
         #endregion
 
         public event PropertyChangedEventHandler PropertyChanged;
 
         protected virtual void OnPropertyChanged([CallerMemberName] string propertyName = null)
         {
-            var handler = this.PropertyChanged;
-            if (handler != null)
+            this.PropertyChanged?.Invoke(this, new PropertyChangedEventArgs(propertyName));
+        }
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposed)
             {
-                handler(this, new PropertyChangedEventArgs(propertyName));
+                if (disposing)
+                {
+                    this.filterTimer?.Dispose();
+                }
+
+                var controller = this.npmController;
```

Comment:
>var controller = this.npmController [](start = 16, length = 35)

The reason not to just set the corresponding property to null is that we don't want to fire a property changed event?

Reply:
yep